src/plugins/data/public/index.ts

```diff
@@ -311,6 +311,8 @@ export {
   SearchStrategyProvider,
   ISearchSource,
   SearchSource,
+  parseSearchSource,
```

Comment:
@lukeelmers This PR is not ready for review from the app arch team yet, but before cleaning up the implementation  I want to make sure the approach is sound. The two utility functions added to the `data` plugin are:
```ts
async function parseSearchSource(
  searchSourceJson: string,
  references: SavedObjectReference[],
  indexPatterns: IndexPatternsContract
): SearchSource;

function serializeSearchSource(searchSource: ISearchSource): { searchSourceJSON: string, references: SavedObjectReference[] };
```

Should these be exposed via contract instead (especially because parsing the search source requires the index patterns service, so we could wire that up within the data plugin instead of passing it in from the consumer)?

Reply:
IMO either approach works, though I do agree it would be nice to provide a version that already works with index patterns.

This is a bit subjective, but in general I think we often sacrifice a clean contract by statically exporting so many things, so I’m personally not opposed to putting both in the lifecycle contract (even the static one). If things change in the future, it is easier to move something from stateful > static than it is to go from static > stateful. 

Plus, from a DX perspective it feels like this would be nice living alongside the other SearchSource stuff in the lifecycle contract  

That said, another way we’ve addressed these scenarios is to build it to be stateless/static, but then offer a “prewired” version in the lifecycle contract. (So basically combining both approaches). In the rare case that someone prefers the static version which requires additional config, they can use it.

Reply:
i prefer having it on the contract but i agree with luke that both approaches work